// viem/dist/cjs/clients/decorators/public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.publicActions = void 0;
const index_js_1 = require("../../actions/ens/index.js");
const index_js_2 = require("../../actions/public/index.js");
const publicActions = (client) => ({
    call: (args) => (0, index_js_2.call)(client, args),
    createBlockFilter: () => (0, index_js_2.createBlockFilter)(client),
    createContractEventFilter: (args) => (0, index_js_2.createContractEventFilter)(client, args),
    createEventFilter: (args) => (0, index_js_2.createEventFilter)(client, args),
    createPendingTransactionFilter: () => (0, index_js_2.createPendingTransactionFilter)(client),
    estimateContractGas: (args) => (0, index_js_2.estimateContractGas)(client, args),
    estimateGas: (args) => (0, index_js_2.estimateGas)(client, args),
    getBalance: (args) => (0, index_js_2.getBalance)(client, args),
    getBlock: (args) => (0, index_js_2.getBlock)(client, args),
    getBlockNumber: (args) => (0, index_js_2.getBlockNumber)(client, args),
    getBlockTransactionCount: (args) => (0, index_js_2.getBlockTransactionCount)(client, args),
    getBytecode: (args) => (0, index_js_2.getBytecode)(client, args),
    getChainId: () => (0, index_js_2.getChainId)(client),
    getEnsAddress: (args) => (0, index_js_1.getEnsAddress)(client, args),
    getEnsAvatar: (args) => (0, index_js_1.getEnsAvatar)(client, args),
    getEnsName: (args) => (0, index_js_1.getEnsName)(client, args),
    getEnsResolver: (args) => (0, index_js_1.getEnsResolver)(client, args),
    getEnsText: (args) => (0, index_js_1.getEnsText)(client, args),
    getFeeHistory: (args) => (0, index_js_2.getFeeHistory)(client, args),
    getFilterChanges: (args) => (0, index_js_2.getFilterChanges)(client, args),
    getFilterLogs: (args) => (0, index_js_2.getFilterLogs)(client, args),
    getGasPrice: () => (0, index_js_2.getGasPrice)(client),
    getLogs: (args) => (0, index_js_2.getLogs)(client, args),
    getStorageAt: (args) => (0, index_js_2.getStorageAt)(client, args),
    getTransaction: (args) => (0, index_js_2.getTransaction)(client, args),
    getTransactionConfirmations: (args) => (0, index_js_2.getTransactionConfirmations)(client, args),
    getTransactionCount: (args) => (0, index_js_2.getTransactionCount)(client, args),
    getTransactionReceipt: (args) => (0, index_js_2.getTransactionReceipt)(client, args),
    multicall: (args) => (0, index_js_2.multicall)(client, args),
    readContract: (args) => (0, index_js_2.readContract)(client, args),
    simulateContract: (args) => (0, index_js_2.simulateContract)(client, args),
    uninstallFilter: (args) => (0, index_js_2.uninstallFilter)(client, args),
    waitForTransactionReceipt: (args) => (0, index_js_2.waitForTransactionReceipt)(client, args),
    watchBlocks: (args) => (0, index_js_2.watchBlocks)(client, args),
    watchBlockNumber: (args) => (0, index_js_2.watchBlockNumber)(client, args),
    watchContractEvent: (args) => (0, index_js_2.watchContractEvent)(client, args),
    watchEvent: (args) => (0, index_js_2.watchEvent)(client, args),
    watchPendingTransactions: (args) => (0, index_js_2.watchPendingTransactions)(client, args),
});
exports.publicActions = publicActions;
