// viem/dist/cjs/clients/decorators/wallet.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.walletActions = void 0;
const index_js_1 = require("../../actions/wallet/index.js");
const walletActions = (client) => ({
    addChain: (args) => (0, index_js_1.addChain)(client, args),
    deployContract: (args) => (0, index_js_1.deployContract)(client, args),
    getAddresses: () => (0, index_js_1.getAddresses)(client),
    getChainId: () => (0, index_js_1.getChainId)(client),
    getPermissions: () => (0, index_js_1.getPermissions)(client),
    requestAddresses: () => (0, index_js_1.requestAddresses)(client),
    requestPermissions: (args) => (0, index_js_1.requestPermissions)(client, args),
    sendTransaction: (args) => (0, index_js_1.sendTransaction)(client, args),
    signMessage: (args) => (0, index_js_1.signMessage)(client, args),
    signTypedData: (args) => (0, index_js_1.signTypedData)(client, args),
    switchChain: (args) => (0, index_js_1.switchChain)(client, args),
    watchAsset: (args) => (0, index_js_1.watchAsset)(client, args),
    writeContract: (args) => (0, index_js_1.writeContract)(client, args),
});
exports.walletActions = walletActions;
