// @bgd-labs/report-engine/dist/chunk-VIOMWIBP.js
// src/renderer/reserves.ts
function renderReserves(reserves) {
  const { added, removed, altered } = Object.entries(reserves).sort(([keyA], [keyB]) => {
    return keyA.localeCompare(keyB, "en-US");
  }).reduce(
    (acc, [key, value]) => {
      if (value.from) {
        acc.removed.push(value.from);
      } else if (value.to) {
        acc.added.push(value.to);
      } else {
        acc.altered.push(value);
      }
      return acc;
    },
    {
      added: [],
      removed: [],
      altered: []
    }
  );
  if (!altered.length && !added.length && !removed.length)
    return "";
  let content = "";
  content += `## Reserves

`;
  if (altered.length) {
    content += `### Reserve altered

`;
    for (const reserve of altered) {
      content += `| key | value |
`;
      content += `| --- | --- |
`;
      Object.keys(reserve).map((key) => {
        content += `| ${key} | ~~${reserve[key].from}~~${reserve[key].to} |
`;
      });
      content += `

`;
    }
  }
  if (added.length) {
    content += `### Reserve added

`;
    for (const reserve of added) {
      content += `| key | value |
`;
      content += `| --- | --- |
`;
      Object.keys(reserve).map((key) => {
        content += `| ${key} | ${reserve[key]} |
`;
      });
      content += `

`;
    }
  }
  if (removed.length) {
    content += `### Reserve added

`;
    for (const reserve of removed) {
      content += `| key | value |
`;
      content += `| --- | --- |
`;
      Object.keys(reserve).map((key) => {
        content += `| ${key} | ${reserve[key]} |
`;
      });
      content += `

`;
    }
  }
  return content;
}

// src/utils/diffler.ts
function diffler(obj1, obj2) {
  let diff = {};
  for (const key in obj1) {
    if (obj1.hasOwnProperty(key) && typeof obj1[key] !== "function") {
      const obj1Val = obj1[key];
      const obj2Val = obj2[key];
      if (typeof obj1Val !== typeof obj2Val) {
        diff[key] = {
          from: obj1Val,
          to: obj2Val
        };
        break;
      }
      if (!(key in obj2)) {
        diff[key] = {
          from: obj1Val,
          to: null
          // using null to specify that the value is empty in obj2
        };
      } else if (typeof obj1Val === "object") {
        var tempDiff = diffler(
          obj1Val,
          obj2Val
        );
        if (Object.keys(tempDiff).length > 0) {
          if (tempDiff) {
            diff[key] = tempDiff;
          }
        }
      } else if (obj1Val !== obj2Val) {
        diff[key] = {
          from: obj1Val,
          to: obj2Val
        };
      }
    }
  }
  for (const key in obj2) {
    if (obj2.hasOwnProperty(key) && typeof obj2[key] !== "function") {
      if (obj1 === null) {
        diff[key] = {
          from: obj1,
          to: obj2[key]
        };
        break;
      }
      const obj2Val = obj2[key];
      if (!(key in obj1)) {
        if (!diff) {
          diff = {};
        }
        diff[key] = {
          from: null,
          to: obj2Val
        };
      }
    }
  }
  return diff;
}

// src/diffReports.ts
function diffReports(pre, post) {
  const diff = diffler(pre, post);
  let content = renderReserves(diff.reserves);
  content += `### Raw diff

\`\`\`json
${JSON.stringify(
    diff,
    null,
    2
  )}
\`\`\``;
  return content;
}

export {
  diffReports
};
/**
 * @dev this is a modified version of the `diffler` npm package.
 * The modification adds exact types.
 * @license MIT https://github.com/omgaz/diffler
 * Author: Gary Chisholm @omgaz
 */
