// viem/dist/cjs/actions/wallet/sendTransaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendTransaction = void 0;
const index_js_1 = require("../../errors/index.js");
const index_js_2 = require("../../utils/index.js");
const index_js_3 = require("../public/index.js");
async function sendTransaction(client, args) {
    const { account: account_ = client.account, chain = client.chain, accessList, data, gas, gasPrice, maxFeePerGas, maxPriorityFeePerGas, nonce, to, value, ...rest } = args;
    if (!account_)
        throw new index_js_1.AccountNotFoundError({
            docsPath: '/docs/actions/wallet/sendTransaction',
        });
    const account = (0, index_js_2.parseAccount)(account_);
    try {
        (0, index_js_2.assertRequest)(args);
        const chainId = await (0, index_js_3.getChainId)(client);
        if (chain !== null && chainId !== chain?.id) {
            if (!chain)
                throw new index_js_1.ChainNotFoundError();
            throw new index_js_1.ChainMismatchError({ chain, currentChainId: chainId });
        }
        if (account.type === 'local') {
            const request = await (0, index_js_2.prepareRequest)(client, {
                account,
                accessList,
                chain,
                data,
                gas,
                gasPrice,
                maxFeePerGas,
                maxPriorityFeePerGas,
                nonce,
                to,
                value,
                ...rest,
            });
            const signedRequest = (await account.signTransaction({
                chainId,
                ...request,
            }));
            return await client.request({
                method: 'eth_sendRawTransaction',
                params: [signedRequest],
            });
        }
        const formatter = chain?.formatters?.transactionRequest;
        const request = (0, index_js_2.format)({
            accessList,
            data,
            from: account.address,
            gas,
            gasPrice,
            maxFeePerGas,
            maxPriorityFeePerGas,
            nonce,
            to,
            value,
            ...(0, index_js_2.extract)(rest, { formatter }),
        }, {
            formatter: formatter || index_js_2.formatTransactionRequest,
        });
        return await client.request({
            method: 'eth_sendTransaction',
            params: [request],
        });
    }
    catch (err) {
        throw (0, index_js_2.getTransactionError)(err, {
            ...args,
            account,
            chain: args.chain || undefined,
        });
    }
}
exports.sendTransaction = sendTransaction;
