// viem/dist/esm/actions/public/estimateGas.js
import { AccountNotFoundError } from '../../errors/index.js';
import { assertRequest, extract, format, formatTransactionRequest, getEstimateGasError, numberToHex, parseAccount, prepareRequest, } from '../../utils/index.js';
export async function estimateGas(client, args) {
    if (!args.account)
        throw new AccountNotFoundError({
            docsPath: '/docs/actions/public/estimateGas',
        });
    const account = parseAccount(args.account);
    try {
        const { accessList, blockNumber, blockTag = 'latest', data, gas, gasPrice, maxFeePerGas, maxPriorityFeePerGas, nonce, to, value, ...rest } = account.type === 'local' ? await prepareRequest(client, args) : args;
        const blockNumberHex = blockNumber ? numberToHex(blockNumber) : undefined;
        assertRequest(args);
        const formatter = client.chain?.formatters?.transactionRequest;
        const request = format({
            from: account.address,
            accessList,
            data,
            gas,
            gasPrice,
            maxFeePerGas,
            maxPriorityFeePerGas,
            nonce,
            to,
            value,
            ...extract(rest, { formatter }),
        }, {
            formatter: formatter || formatTransactionRequest,
        });
        const balance = await client.request({
            method: 'eth_estimateGas',
            params: [request, blockNumberHex || blockTag],
        });
        return BigInt(balance);
    }
    catch (err) {
        throw getEstimateGasError(err, {
            ...args,
            account,
            chain: client.chain,
        });
    }
}
