// viem/dist/esm/actions/public/simulateContract.js
import { decodeFunctionResult, encodeFunctionData, getContractError, parseAccount, } from '../../utils/index.js';
import { call } from './call.js';
export async function simulateContract(client, { abi, address, args, functionName, ...callRequest }) {
    const account = callRequest.account
        ? parseAccount(callRequest.account)
        : undefined;
    const calldata = encodeFunctionData({
        abi,
        args,
        functionName,
    });
    try {
        const { data } = await call(client, {
            batch: false,
            data: calldata,
            to: address,
            ...callRequest,
        });
        const result = decodeFunctionResult({
            abi,
            args,
            functionName,
            data: data || '0x',
        });
        return {
            result,
            request: {
                abi,
                address,
                args,
                functionName,
                ...callRequest,
            },
        };
    }
    catch (err) {
        throw getContractError(err, {
            abi: abi,
            address,
            args,
            docsPath: '/docs/contract/simulateContract',
            functionName,
            sender: account?.address,
        });
    }
}
