// viem/dist/cjs/utils/abi/encodePacked.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.encodePacked = void 0;
const index_js_1 = require("../../errors/index.js");
const index_js_2 = require("../address/index.js");
const index_js_3 = require("../data/index.js");
const index_js_4 = require("../encoding/index.js");
const regex_js_1 = require("../regex.js");
function encodePacked(types, values) {
    if (types.length !== values.length)
        throw new index_js_1.AbiEncodingLengthMismatchError({
            expectedLength: types.length,
            givenLength: values.length,
        });
    const data = [];
    for (let i = 0; i < types.length; i++) {
        const type = types[i];
        const value = values[i];
        data.push(encode(type, value));
    }
    return (0, index_js_3.concat)(data);
}
exports.encodePacked = encodePacked;
function encode(type, value, isArray = false) {
    if (type === 'address') {
        const address = value;
        if (!(0, index_js_2.isAddress)(address))
            throw new index_js_1.InvalidAddressError({ address });
        return (0, index_js_3.pad)(address.toLowerCase(), {
            size: isArray ? 32 : null,
        });
    }
    if (type === 'string')
        return (0, index_js_4.stringToHex)(value);
    if (type === 'bytes')
        return value;
    if (type === 'bool')
        return (0, index_js_3.pad)((0, index_js_4.boolToHex)(value), { size: isArray ? 32 : 1 });
    const intMatch = type.match(regex_js_1.integerRegex);
    if (intMatch) {
        const [_type, baseType, bits = '256'] = intMatch;
        const size = parseInt(bits) / 8;
        return (0, index_js_4.numberToHex)(value, {
            size: isArray ? 32 : size,
            signed: baseType === 'int',
        });
    }
    const bytesMatch = type.match(regex_js_1.bytesRegex);
    if (bytesMatch) {
        const [_type, size] = bytesMatch;
        if (parseInt(size) !== (value.length - 2) / 2)
            throw new index_js_1.BytesSizeMismatchError({
                expectedSize: parseInt(size),
                givenSize: (value.length - 2) / 2,
            });
        return (0, index_js_3.pad)(value, { dir: 'right', size: isArray ? 32 : null });
    }
    const arrayMatch = type.match(regex_js_1.arrayRegex);
    if (arrayMatch && Array.isArray(value)) {
        const [_type, childType] = arrayMatch;
        const data = [];
        for (let i = 0; i < value.length; i++) {
            data.push(encode(childType, value[i], true));
        }
        return (0, index_js_3.concat)(data);
    }
    throw new index_js_1.UnsupportedPackedAbiType(type);
}
