// viem/dist/cjs/actions/ens/getEnsName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getEnsName = void 0;
const index_js_1 = require("../../constants/index.js");
const index_js_2 = require("../../errors/index.js");
const index_js_3 = require("../../utils/index.js");
const index_js_4 = require("../../utils/ens/index.js");
const index_js_5 = require("../public/index.js");
async function getEnsName(client, { address, blockNumber, blockTag, universalResolverAddress: universalResolverAddress_, }) {
    let universalResolverAddress = universalResolverAddress_;
    if (!universalResolverAddress) {
        if (!client.chain)
            throw new Error('client chain not configured. universalResolverAddress is required.');
        universalResolverAddress = (0, index_js_3.getChainContractAddress)({
            blockNumber,
            chain: client.chain,
            contract: 'ensUniversalResolver',
        });
    }
    const reverseNode = `${address.toLowerCase().substring(2)}.addr.reverse`;
    try {
        const res = await (0, index_js_5.readContract)(client, {
            address: universalResolverAddress,
            abi: [
                {
                    name: 'reverse',
                    type: 'function',
                    stateMutability: 'view',
                    inputs: [{ type: 'bytes', name: 'reverseName' }],
                    outputs: [
                        { type: 'string', name: 'resolvedName' },
                        { type: 'address', name: 'resolvedAddress' },
                        { type: 'address', name: 'reverseResolver' },
                        { type: 'address', name: 'resolver' },
                    ],
                },
            ],
            functionName: 'reverse',
            args: [(0, index_js_3.toHex)((0, index_js_4.packetToBytes)(reverseNode))],
            blockNumber,
            blockTag,
        });
        return res[0];
    }
    catch (error) {
        if (error instanceof index_js_2.ContractFunctionExecutionError &&
            error.cause.reason === index_js_1.panicReasons[50])
            return null;
        throw error;
    }
}
exports.getEnsName = getEnsName;
