// viem/dist/cjs/ens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.normalize = exports.namehash = exports.labelhash = exports.getEnsText = exports.getEnsResolver = exports.getEnsName = exports.getEnsAvatar = exports.getEnsAddress = void 0;
var index_js_1 = require("./actions/ens/index.js");
Object.defineProperty(exports, "getEnsAddress", { enumerable: true, get: function () { return index_js_1.getEnsAddress; } });
Object.defineProperty(exports, "getEnsAvatar", { enumerable: true, get: function () { return index_js_1.getEnsAvatar; } });
Object.defineProperty(exports, "getEnsName", { enumerable: true, get: function () { return index_js_1.getEnsName; } });
Object.defineProperty(exports, "getEnsResolver", { enumerable: true, get: function () { return index_js_1.getEnsResolver; } });
Object.defineProperty(exports, "getEnsText", { enumerable: true, get: function () { return index_js_1.getEnsText; } });
var index_js_2 = require("./utils/ens/index.js");
Object.defineProperty(exports, "labelhash", { enumerable: true, get: function () { return index_js_2.labelhash; } });
Object.defineProperty(exports, "namehash", { enumerable: true, get: function () { return index_js_2.namehash; } });
var normalize_js_1 = require("./utils/ens/normalize.js");
Object.defineProperty(exports, "normalize", { enumerable: true, get: function () { return normalize_js_1.normalize; } });
