// @bgd-labs/aave-address-book/dist/AaveV2EthereumArc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PERMISSION_MANAGER = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x6FdfafB66d39cD72CFE7984D3Bbcc76632faAb00';
exports.POOL = '0x37D7306019a38Af123e4b245Eb6C28AF552e0bB0';
exports.POOL_CONFIGURATOR = '0x4e1c7865e7BE78A7748724Fa0409e88dc14E67aA';
exports.LENDING_RATE_ORACLE = '0xfA3c34d734fe0106C87917683ca45dffBe3b3B00';
exports.ORACLE = '0xB8a7bc0d13B1f5460513040a97F404b4fea7D2f3';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x71B53fC437cCD988b1b89B1D4605c3c3d0C810ea';
exports.POOL_ADMIN = '0xAce1d11d836cb3F51Ef658FD4D353fFb3c301218';
exports.EMERGENCY_ADMIN = '0x33B09130b035d6D7e57d76fEa0873d9545FA7557';
exports.COLLECTOR = '0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x0000000000000000000000000000000000000000';
exports.EMISSION_MANAGER = '0x0000000000000000000000000000000000000000';
exports.CHAIN_ID = 1;
exports.PERMISSION_MANAGER = '0xF4a1F5fEA79C3609514A417425971FadC10eCfBE';
