// @bgd-labs/aave-address-book/dist/AaveV2Goerli.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FAUCET = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x5E52dEc931FFb32f609681B8438A51c675cc232d';
exports.POOL = '0x4bd5643ac6f66a5237E18bfA7d47cF22f1c9F210';
exports.POOL_CONFIGURATOR = '0x88B1D3d97656De3Ec44FEDDfa109AF7fb8C2837D';
exports.LENDING_RATE_ORACLE = '0x76aFA2b6C29E1B277A3BB1CD320b2756c1674c91';
exports.ORACLE = '0x2cb0d5755436ED904D7D0fbBACc6176286c55667';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x0000000000000000000000000000000000000000';
exports.POOL_ADMIN = '0x77c45699A715A64A7a7796d5CEe884cf617D5254';
exports.EMERGENCY_ADMIN = '0x77c45699A715A64A7a7796d5CEe884cf617D5254';
exports.COLLECTOR = '0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x0000000000000000000000000000000000000000';
exports.EMISSION_MANAGER = '0x0000000000000000000000000000000000000000';
exports.CHAIN_ID = 5;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x3465454D658019f8A0eABD3bC61d2d1Dd3a0735F';
exports.WETH_GATEWAY = '0x3bd3a20Ac9Ff1dda1D99C0dFCE6D65C4960B3627';
exports.WALLET_BALANCE_PROVIDER = '0xf1E4A6E7FA07421FD5139Ba0848290A27e22db7f';
exports.UI_POOL_DATA_PROVIDER = '0xaaa2872d1F7f5ceb630Cb736BcA34Ff1e121992b';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xA2E05bE2090b3658A264bdf1C39387f5Dba367Ec';
exports.FAUCET = '0x681860075529352da2C94082Eb66c59dF958e89C';
