// @bgd-labs/aave-address-book/dist/AaveGovernanceV2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CROSSCHAIN_FORWARDER_ARBITRUM = exports.CROSSCHAIN_FORWARDER_OPTIMISM = exports.CROSSCHAIN_FORWARDER_POLYGON = exports.ARBITRUM_BRIDGE_EXECUTOR = exports.OPTIMISM_BRIDGE_EXECUTOR = exports.POLYGON_BRIDGE_EXECUTOR = exports.ARC_TIMELOCK = exports.LONG_EXECUTOR = exports.SHORT_EXECUTOR = exports.GOV_STRATEGY = exports.GOV = void 0;
exports.GOV = '0xEC568fffba86c094cf06b22134B23074DFE2252c';
exports.GOV_STRATEGY = '0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e';
exports.SHORT_EXECUTOR = '0xEE56e2B3D491590B5b31738cC34d5232F378a8D5';
exports.LONG_EXECUTOR = '0x79426A1c24B2978D90d7A5070a46C65B07bC4299';
exports.ARC_TIMELOCK = '0xAce1d11d836cb3F51Ef658FD4D353fFb3c301218';
exports.POLYGON_BRIDGE_EXECUTOR = '0xdc9A35B16DB4e126cFeDC41322b3a36454B1F772';
exports.OPTIMISM_BRIDGE_EXECUTOR = '0x7d9103572bE58FfE99dc390E8246f02dcAe6f611';
exports.ARBITRUM_BRIDGE_EXECUTOR = '0x7d9103572bE58FfE99dc390E8246f02dcAe6f611';
exports.CROSSCHAIN_FORWARDER_POLYGON = '0x158a6bC04F0828318821baE797f50B0A1299d45b';
exports.CROSSCHAIN_FORWARDER_OPTIMISM = '0x5f5C02875a8e9B5A26fbd09040ABCfDeb2AA6711';
exports.CROSSCHAIN_FORWARDER_ARBITRUM = '0xd1B3E25fD7C8AE7CADDC6F71b461b79CD4ddcFa3';
