// @bgd-labs/aave-address-book/dist/AaveV3GoerliGho.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x4dd5ab8Fb385F2e12aDe435ba7AFA812F1d364D0';
exports.POOL = '0x617Cf26407193E32a771264fB5e9b8f09715CdfB';
exports.POOL_CONFIGURATOR = '0x04eA6B2a9E82Bd58bC95B8eA2b90496356dD969F';
exports.ORACLE = '0xcb601629B36891c43943e3CDa2eB18FAc38B5c4e';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0xB7d8ff9949dB06D8387C28332045b8F734641755';
exports.ACL_MANAGER = '0x9C94757E231AdF6c90f89259DFA0841a1bf8172f';
exports.ACL_ADMIN = '0x2892e37624Ec31CC42502f297821109700270971';
exports.COLLECTOR = '0x1A47958e231848C664863D213bC27b018934477D';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x8688FEF353f4940061b4893d563de1C487Aa92Fd';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0x877880d5E01b483add2f3223B709a26b622F0852';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x46Dbc9aeF756a594a61fcbB35902F24103044cBf';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0xAc08Af18C2E9b7B62facacd048d2F89a25024474';
exports.CHAIN_ID = 5;
exports.EMISSION_MANAGER = '0xC9f3b10c05a3e2a49F8165AC29b9275f85DE06cD';
exports.WETH_GATEWAY = '0x9c402E3b0D123323F0FCed781b8184Ec7E02Dd31';
exports.FAUCET = '0x1265305F033156bbF8Ba54fE45DD5685BEc4Cc44';
exports.WALLET_BALANCE_PROVIDER = '0x03C8d0c46834921c4468C15A03E5d76Ae5CA3133';
exports.UI_POOL_DATA_PROVIDER = '0x3De59b6901e7Ad0A19621D49C5b52cC9a4977e52';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xF67B25977cEFf3563BF7F24A531D6CEAe6870a9d';
