// @bgd-labs/aave-address-book/dist/AaveV2Polygon.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.MIGRATION_HELPER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.SWAP_COLLATERAL_ADAPTER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xd05e3E715d945B59290df0ae8eF85c1BdB684744';
exports.POOL = '0x8dFf5E27EA6b7AC08EbFdf9eB090F32ee9a30fcf';
exports.POOL_CONFIGURATOR = '0x26db2B833021583566323E3b8985999981b9F1F3';
exports.LENDING_RATE_ORACLE = '0x17F73aEaD876CC4059089ff815EDA37052960dFB';
exports.ORACLE = '0x0229F777B0fAb107F9591a41d5F02E4e98dB6f2d';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x7551b5D2763519d4e37e8B81929D336De671d46d';
exports.POOL_ADMIN = '0xdc9A35B16DB4e126cFeDC41322b3a36454B1F772';
exports.EMERGENCY_ADMIN = '0x1450F2898D6bA2710C98BE9CAF3041330eD5ae58';
exports.COLLECTOR = '0xe8599F3cc5D38a9aD6F3684cd5CEa72f10Dbc383';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x357D51124f59836DeD84c8a1730D72B749d8BC23';
exports.EMISSION_MANAGER = '0x2bB25175d9B0F8965780209EB558Cc3b56cA6d32';
exports.CHAIN_ID = 137;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x3ac4e9aa29940770aeC38fe853a4bbabb2dA9C19';
exports.WETH_GATEWAY = '0xAeBF56223F044a73A513FAD7E148A9075227eD9b';
exports.SWAP_COLLATERAL_ADAPTER = '0x35784a624D4FfBC3594f4d16fA3801FeF063241c';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0xE84cF064a0a65290Ae5673b500699f3753063936';
exports.MIGRATION_HELPER = '0x3db487975aB1728DB5787b798866c2021B24ec52';
exports.WALLET_BALANCE_PROVIDER = '0x34aa032bC416Cf2CdC45c0C8f065b1F19463D43e';
exports.UI_POOL_DATA_PROVIDER = '0x204f2Eb81D996729829debC819f7992DCEEfE7b1';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x645654D59A5226CBab969b1f5431aA47CBf64ab8';
