// @bgd-labs/aave-address-book/dist/AaveV3Arbitrum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.STATIC_A_TOKEN_FACTORY = exports.L2_ENCODER = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.LISTING_ENGINE = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.RATES_FACTORY = exports.SWAP_COLLATERAL_ADAPTER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xa97684ead0e402dC232d5A977953DF7ECBaB3CDb';
exports.POOL = '0x794a61358D6845594F94dc1DB02A252b5b4814aD';
exports.POOL_CONFIGURATOR = '0x8145eddDf43f50276641b55bd3AD95944510021E';
exports.ORACLE = '0xb56c2F0B653B2e0b10C9b928C8580Ac5Df02C7C7';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x69FA688f1Dc47d4B5d8029D5a35FB7a548310654';
exports.ACL_MANAGER = '0xa72636CbcAa8F5FF95B2cc47F3CDEe83F3294a0B';
exports.ACL_ADMIN = '0x7d9103572bE58FfE99dc390E8246f02dcAe6f611';
exports.COLLECTOR = '0x053D55f9B5AF8694c503EB288a1B7E552f590710';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x929EC64c34a17401F460460D4B9390518E5B473e';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xa5ba6E5EC19a1Bf23C857991c857dB62b2Aa187B';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x81387c40EB75acB02757C1Ae55D5936E78c9dEd3';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x52A1CeB68Ee6b7B5D13E0376A1E0E4423A8cE26e';
exports.CHAIN_ID = 42161;
exports.EMISSION_MANAGER = '0x048f2228D7Bf6776f99aB50cB1b1eaB4D1d4cA73';
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x770ef9f4fe897e59daCc474EF11238303F9552b6';
exports.WETH_GATEWAY = '0xB5Ee21786D28c5Ba61661550879475976B707099';
exports.SWAP_COLLATERAL_ADAPTER = '0xb20bd14259089e63d1f49d8d887ABA783Ce5dAc2';
exports.RATES_FACTORY = '0xcC47c4Fe1F7f29ff31A8b62197023aC8553C7896';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0xB0526BFb4047aE1147DC7caAF3F1653904C2D568';
exports.LISTING_ENGINE = '0x0EfdfC1A940DE4E7E6acC9Bb801481f81B17fd20';
exports.WALLET_BALANCE_PROVIDER = '0xBc790382B3686abffE4be14A030A96aC6154023a';
exports.UI_POOL_DATA_PROVIDER = '0x145dE30c929a065582da84Cf96F88460dB9745A7';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xDA67AF3403555Ce0AE3ffC22fDb7354458277358';
exports.L2_ENCODER = '0x9abADECD08572e0eA5aF4d47A9C7984a5AA503dC';
exports.STATIC_A_TOKEN_FACTORY = '0x90b1255a76e847cC92d41C295DeD5Bf2D4F24B3d';
