// @bgd-labs/aave-address-book/dist/AaveV2Fuji.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x7fdC1FdF79BE3309bf82f4abdAD9f111A6590C0f';
exports.POOL = '0x76cc67FF2CC77821A70ED14321111Ce381C2594D';
exports.POOL_CONFIGURATOR = '0x4ceBAFAAcc6Cb26FD90E4cDe138Eb812442bb5f3';
exports.LENDING_RATE_ORACLE = '0x76Ec7c83aCb6af821E61F1DF1E0aBE684Bc904F8';
exports.ORACLE = '0xfa4f5B081632c4709667D467F817C09d9008A46A';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x0668EDE013c1c475724523409b8B6bE633469585';
exports.POOL_ADMIN = '0x1128d177BdaA74Ae68EB06e693f4CbA6BF427a5e';
exports.EMERGENCY_ADMIN = '0x1128d177BdaA74Ae68EB06e693f4CbA6BF427a5e';
exports.COLLECTOR = '0xB45F5C501A22288dfdb897e5f73E189597e09288';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0xa1EF206fb9a8D8186157FC817fCddcC47727ED55';
exports.EMISSION_MANAGER = '0x3b60cABB2C0e9ADe3e364b1F9752342A5D6079e2';
exports.CHAIN_ID = 43113;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x04A6Fa9999E3C807Ee7b6Ca58eFAb93713d405BF';
exports.WETH_GATEWAY = '0x1648C14DbB6ccdd5846969cE23DeEC4C66a03335';
exports.FAUCET = '0x90E5BAc5A98fff59617080848959f44eACB4Cd7B';
exports.WALLET_BALANCE_PROVIDER = '0x3f5A507B33260a3869878B31FB90F04F451d28e3';
exports.UI_POOL_DATA_PROVIDER = '0x88b4013f8C50e61ab027Cc253ab9a50663e2dF45';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x9842E5B7b7C6cEDfB1952a388e050582Ff95645b';
