// @bgd-labs/aave-address-book/dist/AaveV3OptimismGoerli.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.L2_ENCODER = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x0b8FAe5f9Bf5a1a5867FB5b39fF4C028b1C2ebA9';
exports.POOL = '0xCAd01dAdb7E97ae45b89791D986470F3dfC256f7';
exports.POOL_CONFIGURATOR = '0x98EC9f3e5A0E5aDB16BAAdEB96a110BceeaC0067';
exports.ORACLE = '0x2366d0cE3f44D81f7b2D40C64288b5eAA7593049';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x861d7d4A46C63b92461631CC77a9f2aeAcFfA10d';
exports.ACL_MANAGER = '0x3e965db7b1BaA260B65208e3F508eD84344ebd75';
exports.ACL_ADMIN = '0xaDdfe0b2342800ebD67C30d1c2Bd479E4D498BD5';
exports.COLLECTOR = '0x026E3e3363843e16e3D6d21e068c981A4F55e5d2';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x062BB55A42875366DB1B7D227B73621C33a6cB6b';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0x7675E8C3e91A11D721D0292331c5ee28ed8996ee';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0xF8707057529639A3da9D951054DE89f66d01B3e9';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0xf12460DD042d7F143c6c5ab7A0C5CeA24F7a20b7';
exports.CHAIN_ID = 420;
exports.EMISSION_MANAGER = '0x936fF44bb41Fe8d6c3028A016D3255cB3296ECA0';
exports.WETH_GATEWAY = '0x6f7f2440006221F893c587b88f01afc42B6F8d2e';
exports.FAUCET = '0x777A5810352302A2D6d79d5B7323237c467845d9';
exports.WALLET_BALANCE_PROVIDER = '0xb463057Eb60E1575e2a69aa17C63CCd2F3161a5f';
exports.UI_POOL_DATA_PROVIDER = '0x9277eFbB991536a98a1aA8b735E9D26d887104C1';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x4157398c5abB5211F51F5B551E3e240c5568dbD4';
exports.L2_ENCODER = '0x14AA09449fac437b5c0110614be2C08610e38f62';
