// @bgd-labs/aave-address-book/dist/AaveV3Ethereum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.STATIC_A_TOKEN_FACTORY = exports.DELEGATION_AWARE_A_TOKEN_IMPL_REV_1 = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.LISTING_ENGINE = exports.SWAP_COLLATERAL_ADAPTER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.RATES_FACTORY = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x2f39d218133AFaB8F2B819B1066c7E434Ad94E9e';
exports.POOL = '0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2';
exports.POOL_CONFIGURATOR = '0x64b761D848206f447Fe2dd461b0c635Ec39EbB27';
exports.ORACLE = '0x54586bE62E3c3580375aE3723C145253060Ca0C2';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x7B4EB56E7CD4b454BA8ff71E4518426369a138a3';
exports.ACL_MANAGER = '0xc2aaCf6553D20d1e9d78E365AAba8032af9c85b0';
exports.ACL_ADMIN = '0xEE56e2B3D491590B5b31738cC34d5232F378a8D5';
exports.COLLECTOR = '0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x8164Cc65827dcFe994AB23944CBC90e0aa80bFcb';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0x7EfFD7b47Bfd17e52fB7559d3f924201b9DbfF3d';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0xaC725CB59D16C81061BDeA61041a8A5e73DA9EC6';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x15C5620dfFaC7c7366EED66C20Ad222DDbB1eD57';
exports.CHAIN_ID = 1;
exports.EMISSION_MANAGER = '0x223d844fc4B006D67c0cDbd39371A9F73f69d974';
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0xbaA999AC55EAce41CcAE355c77809e68Bb345170';
exports.WETH_GATEWAY = '0xD322A49006FC828F9B5B37Ab215F99B4E5caB19C';
exports.RATES_FACTORY = '0xcC47c4Fe1F7f29ff31A8b62197023aC8553C7896';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0x1809f186D680f239420B56948C58F8DbbCdf1E18';
exports.SWAP_COLLATERAL_ADAPTER = '0x872fBcb1B582e8Cd0D0DD4327fBFa0B4C2730995';
exports.LISTING_ENGINE = '0xE202F2fc4b6A37Ba53cfD15bE42a762A645FCA07';
exports.WALLET_BALANCE_PROVIDER = '0xC7be5307ba715ce89b152f3Df0658295b3dbA8E2';
exports.UI_POOL_DATA_PROVIDER = '0x91c0eA31b49B69Ea18607702c5d9aC360bf3dE7d';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x162A7AC02f547ad796CA549f757e2b8d1D9b10a6';
exports.DELEGATION_AWARE_A_TOKEN_IMPL_REV_1 = '0x21714092D90c7265F52fdfDae068EC11a23C6248';
exports.STATIC_A_TOKEN_FACTORY = '0x90b1255a76e847cC92d41C295DeD5Bf2D4F24B3d';
