// @bgd-labs/aave-address-book/dist/AaveV2EthereumAMM.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xAcc030EF66f9dFEAE9CbB0cd1B25654b82cFA8d5';
exports.POOL = '0x7937D4799803FbBe595ed57278Bc4cA21f3bFfCB';
exports.POOL_CONFIGURATOR = '0x23A875eDe3F1030138701683e42E9b16A7F87768';
exports.LENDING_RATE_ORACLE = '0x8A32f49FFbA88aba6EFF96F45D8BD1D4b3f35c7D';
exports.ORACLE = '0xA50ba011c48153De246E5192C8f9258A2ba79Ca9';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x0000000000000000000000000000000000000000';
exports.POOL_ADMIN = '0xEE56e2B3D491590B5b31738cC34d5232F378a8D5';
exports.EMERGENCY_ADMIN = '0xB9062896ec3A615a4e4444DF183F0531a77218AE';
exports.COLLECTOR = '0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x0000000000000000000000000000000000000000';
exports.EMISSION_MANAGER = '0x0000000000000000000000000000000000000000';
exports.CHAIN_ID = 1;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x52D306e36E3B6B02c153d0266ff0f85d18BCD413';
exports.WETH_GATEWAY = '0x1C4a4e31231F71Fc34867D034a9E68f6fC798249';
exports.WALLET_BALANCE_PROVIDER = '0x8E8dAd5409E0263a51C0aB5055dA66Be28cFF922';
exports.UI_POOL_DATA_PROVIDER = '0x00e50FAB64eBB37b87df06Aa46b8B35d5f1A4e1A';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xD01ab9a6577E1D84F142e44D49380e23A340387d';
