// @bgd-labs/aave-address-book/dist/AaveV3ScrollAlpha.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.L2_ENCODER = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xCA35Ae4cc948Dae8a1d3C77ed1C5CBC2e73b290D';
exports.POOL = '0x556345b249008Bf2f52B7097ceB5bE798F32dAe9';
exports.POOL_CONFIGURATOR = '0x88A4811E6009ad13EA879261806E5a6071F3F788';
exports.ORACLE = '0x269D5C1854fcA3bede6BFc8935118c99020b13f0';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x36002d10bB238594c58aec104620555a545D517C';
exports.ACL_MANAGER = '0x550CC76fD6C4E4502cB4F79f26ee15f4E4681D57';
exports.ACL_ADMIN = '0x8A52430a0a83d2bA00A88758340e4b640BDfC4FC';
exports.COLLECTOR = '0x500A5dc53fD22bdBBe37bE9cc00cB67F24e233d1';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x318101F73b388C73b5A2Da39B4De54f335D77A0D';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xf7798f65d3d1A1Af1a68d8B4322638559A47bC83';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0xF087700794eb98b193D8f95fFEc0E1a02e195e28';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x2bb98A7F11a81AE07443f192Daae04f815F2c878';
exports.CHAIN_ID = 534353;
exports.EMISSION_MANAGER = '0x41A9A5615a912E47F99e16484880009c7c65707b';
exports.WETH_GATEWAY = '0x410Fda971c841Aaf34be5F5539b40503c7F12AC2';
exports.FAUCET = '0x55530C4E1ADFf14cB5760C6750FeBFbEB57E6753';
exports.WALLET_BALANCE_PROVIDER = '0x2B45F4A959B5Fc8fc22b457424685f17eDae4592';
exports.UI_POOL_DATA_PROVIDER = '0x45dFc7A61AD24918C9315733223fD1e55E9B2B59';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x109e475c0Bf9E525dABB65A4Aec07c1e65100a99';
exports.L2_ENCODER = '0x997a8208902e1259dDf676Eb37FeD31A2f77110B';
