// @bgd-labs/aave-address-book/dist/AaveV3Avalanche.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.STATIC_A_TOKEN_FACTORY = exports.PROOF_OF_RESERVE_AGGREGATOR = exports.PROOF_OF_RESERVE = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.LISTING_ENGINE = exports.SWAP_COLLATERAL_ADAPTER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.RATES_FACTORY = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xa97684ead0e402dC232d5A977953DF7ECBaB3CDb';
exports.POOL = '0x794a61358D6845594F94dc1DB02A252b5b4814aD';
exports.POOL_CONFIGURATOR = '0x8145eddDf43f50276641b55bd3AD95944510021E';
exports.ORACLE = '0xEBd36016B3eD09D4693Ed4251c67Bd858c3c7C9C';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x69FA688f1Dc47d4B5d8029D5a35FB7a548310654';
exports.ACL_MANAGER = '0xa72636CbcAa8F5FF95B2cc47F3CDEe83F3294a0B';
exports.ACL_ADMIN = '0xa35b76E4935449E33C56aB24b23fcd3246f13470';
exports.COLLECTOR = '0x5ba7fd868c40c16f7aDfAe6CF87121E13FC2F7a0';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x929EC64c34a17401F460460D4B9390518E5B473e';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xa5ba6E5EC19a1Bf23C857991c857dB62b2Aa187B';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x81387c40EB75acB02757C1Ae55D5936E78c9dEd3';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x52A1CeB68Ee6b7B5D13E0376A1E0E4423A8cE26e';
exports.CHAIN_ID = 43114;
exports.EMISSION_MANAGER = '0x048f2228D7Bf6776f99aB50cB1b1eaB4D1d4cA73';
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x770ef9f4fe897e59daCc474EF11238303F9552b6';
exports.WETH_GATEWAY = '0x6F143FE2F7B02424ad3CaD1593D6f36c0Aab69d7';
exports.RATES_FACTORY = '0xDd81E6F85358292075B78fc8D5830BE8434aF8BA';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0x4fB0Ba43a5415f312cf5FA10d16Ff6dbA697d9aA';
exports.SWAP_COLLATERAL_ADAPTER = '0x00e39085C877328619FDB2a47d995c516a5fA869';
exports.LISTING_ENGINE = '0x49581e5575F49263f556b91daf8fb41D7854D94B';
exports.WALLET_BALANCE_PROVIDER = '0xBc790382B3686abffE4be14A030A96aC6154023a';
exports.UI_POOL_DATA_PROVIDER = '0xF71DBe0FAEF1473ffC607d4c555dfF0aEaDb878d';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x265d414f80b0fca9505710e6F16dB4b67555D365';
exports.PROOF_OF_RESERVE = '0xab22988D93d5F942fC6B6c6Ea285744809D1d9Cc';
exports.PROOF_OF_RESERVE_AGGREGATOR = '0x80f2c02224a2E548FC67c0bF705eBFA825dd5439';
exports.STATIC_A_TOKEN_FACTORY = '0xcC47c4Fe1F7f29ff31A8b62197023aC8553C7896';
