// @bgd-labs/aave-address-book/dist/AaveV3Fantom.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.SWAP_COLLATERAL_ADAPTER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xa97684ead0e402dC232d5A977953DF7ECBaB3CDb';
exports.POOL = '0x794a61358D6845594F94dc1DB02A252b5b4814aD';
exports.POOL_CONFIGURATOR = '0x8145eddDf43f50276641b55bd3AD95944510021E';
exports.ORACLE = '0xfd6f3c1845604C8AE6c6E402ad17fb9885160754';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x69FA688f1Dc47d4B5d8029D5a35FB7a548310654';
exports.ACL_MANAGER = '0xa72636CbcAa8F5FF95B2cc47F3CDEe83F3294a0B';
exports.ACL_ADMIN = '0x39CB97b105173b56b5a2b4b33AD25d6a50E6c949';
exports.COLLECTOR = '0xBe85413851D195fC6341619cD68BfDc26a25b928';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x929EC64c34a17401F460460D4B9390518E5B473e';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xa5ba6E5EC19a1Bf23C857991c857dB62b2Aa187B';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x81387c40EB75acB02757C1Ae55D5936E78c9dEd3';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x52A1CeB68Ee6b7B5D13E0376A1E0E4423A8cE26e';
exports.CHAIN_ID = 250;
exports.EMISSION_MANAGER = '0x048f2228D7Bf6776f99aB50cB1b1eaB4D1d4cA73';
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x770ef9f4fe897e59daCc474EF11238303F9552b6';
exports.WETH_GATEWAY = '0x1DcDA4de2Bf6c7AD9a34788D22aE6b7d55016e1f';
exports.SWAP_COLLATERAL_ADAPTER = '0xE387c6053CE8EC9f8C3fa5cE085Af73114a695d3';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0x1408401B2A7E28cB747b3e258D0831Fc926bAC51';
exports.WALLET_BALANCE_PROVIDER = '0xBc790382B3686abffE4be14A030A96aC6154023a';
exports.UI_POOL_DATA_PROVIDER = '0xddf65434502E459C22263BE2ed7cF0f1FaFD44c0';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x67Da261c14fd94cE7fDd77a0A8476E5b244089A9';
