// @bgd-labs/aave-address-book/dist/AaveV3FantomTestnet.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xC809bea009Ca8DAA680f6A1c4Ca020D550210736';
exports.POOL = '0x95b1B6470eAF8cC4A03d2D44C6b54eBB8ede8C30';
exports.POOL_CONFIGURATOR = '0x91ce34267F11EcB54b2601Ed1C43188cE465dabB';
exports.ORACLE = '0xDd6BCF7EF3dbA79b03D61De36Cc292661c664efD';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x221b58772526669172acCA8B68f6905086c81569';
exports.ACL_MANAGER = '0xbB27a8D0D19fB0c43364Bd26AEB8Fc131F4dA40F';
exports.ACL_ADMIN = '0xaDdfe0b2342800ebD67C30d1c2Bd479E4D498BD5';
exports.COLLECTOR = '0xE4A880b56B4790632753c7393cC51FefFd965678';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0xa45B99c552a2D576B272cc9bFbEB131427ae5148';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xd116069eaBD82DA3A18CA9c5231c1DbB3279Dc0b';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x7074f39fb7A91C251798DAF614dB4e9893c89349';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x7533eACa1cfea1Ade1df6D3fa662E98CFC85cBB9';
exports.CHAIN_ID = 4002;
exports.EMISSION_MANAGER = '0xAf2E5b5cf4aCae5E670eE9619eEd7B90669215f5';
exports.WETH_GATEWAY = '0x87770f04Bbece8092d777860907798138825f303';
exports.FAUCET = '0x77523cB4402d241e324Bcf1EcEa91C4f63033B1b';
exports.WALLET_BALANCE_PROVIDER = '0x4E2e1F992A2ba1137fB6e1FcfbEdcaC95cA788e5';
exports.UI_POOL_DATA_PROVIDER = '0x9a00043F98941DD4e02E1c7e78676df64F5e37a6';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xFBBdDFfFFcFBD55a6DF325d2be47077875Ef9eB9';
