// @bgd-labs/aave-address-book/dist/AaveV2Mumbai.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x178113104fEcbcD7fF8669a0150721e231F0FD4B';
exports.POOL = '0x9198F13B08E299d85E096929fA9781A1E3d5d827';
exports.POOL_CONFIGURATOR = '0xc3c37E2aA3dc66464fa3C29ce2a6EC85beFC45e1';
exports.LENDING_RATE_ORACLE = '0xC661e1445F9a8E5FD3C3dbCa0A0A2e8CBc79725D';
exports.ORACLE = '0xC365C653f7229894F93994CD0b30947Ab69Ff1D5';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0xFA3bD19110d986c5e5E9DD5F69362d05035D045B';
exports.POOL_ADMIN = '0x943E44157dC0302a5CEb172374d1749018a00994';
exports.EMERGENCY_ADMIN = '0x943E44157dC0302a5CEb172374d1749018a00994';
exports.COLLECTOR = '0x943E44157dC0302a5CEb172374d1749018a00994';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0xd41aE58e803Edf4304334acCE4DC4Ec34a63C644';
exports.EMISSION_MANAGER = '0x943E44157dC0302a5CEb172374d1749018a00994';
exports.CHAIN_ID = 80001;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0xE6ef11C967898F9525D550014FDEdCFAB63536B5';
exports.WETH_GATEWAY = '0xee9eE614Ad26963bEc1Bec0D2c92879ae1F209fA';
exports.FAUCET = '0x0b3C23243106A69449e79C14c58BB49E358f9B10';
exports.WALLET_BALANCE_PROVIDER = '0xEe7c0172c200e12AFEa3C34837052ec52F3f367A';
exports.UI_POOL_DATA_PROVIDER = '0xb36a91b1deF63B603896290F6a888c774328519A';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x070a7D8F4d7A7A87452C5BaBaB3158e08411907E';
