// @bgd-labs/aave-address-book/dist/AaveV2Avalanche.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROOF_OF_RESERVE_AGGREGATOR = exports.PROOF_OF_RESERVE = exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.MIGRATION_HELPER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.SWAP_COLLATERAL_ADAPTER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xb6A86025F0FE1862B372cb0ca18CE3EDe02A318f';
exports.POOL = '0x4F01AeD16D97E3aB5ab2B501154DC9bb0F1A5A2C';
exports.POOL_CONFIGURATOR = '0x230B618aD4C475393A7239aE03630042281BD86e';
exports.LENDING_RATE_ORACLE = '0xc34254642B504484465F38Cb1CC396d45a9c7c80';
exports.ORACLE = '0xdC336Cd4769f4cC7E9d726DA53e6d3fC710cEB89';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x65285E9dfab318f57051ab2b139ccCf232945451';
exports.POOL_ADMIN = '0xa35b76E4935449E33C56aB24b23fcd3246f13470';
exports.EMERGENCY_ADMIN = '0xa35b76E4935449E33C56aB24b23fcd3246f13470';
exports.COLLECTOR = '0x467b92aF281d14cB6809913AD016a607b5ba8A36';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x01D83Fe6A10D2f2B7AF17034343746188272cAc9';
exports.EMISSION_MANAGER = '0x5CfCd7E6D055Ba4f7B998914336254aDE3F69f26';
exports.CHAIN_ID = 43114;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x4235E22d9C3f28DCDA82b58276cb6370B01265C2';
exports.WETH_GATEWAY = '0xC27d4dBefc2C0CE57916a699971b58a3BD9C7d5b';
exports.SWAP_COLLATERAL_ADAPTER = '0x2EcF2a2e74B19Aab2a62312167aFF4B78E93B6C5';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0x935b362EE3E1f342cc48118C528AAbee5118F6e6';
exports.MIGRATION_HELPER = '0xf50a080aC535e531EC33cC05b227E910De2fb1fA';
exports.WALLET_BALANCE_PROVIDER = '0x73e4898a1Bfa9f710B6A6AB516403A6299e01fc6';
exports.UI_POOL_DATA_PROVIDER = '0x00e50FAB64eBB37b87df06Aa46b8B35d5f1A4e1A';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x11979886A6dBAE27D7a72c49fCF3F23240D647bF';
exports.PROOF_OF_RESERVE = '0x7fc3FCb14eF04A48Bb0c12f0c39CD74C249c37d8';
exports.PROOF_OF_RESERVE_AGGREGATOR = '0x80f2c02224a2E548FC67c0bF705eBFA825dd5439';
