// @bgd-labs/aave-address-book/dist/AaveV3Harmony.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xa97684ead0e402dC232d5A977953DF7ECBaB3CDb';
exports.POOL = '0x794a61358D6845594F94dc1DB02A252b5b4814aD';
exports.POOL_CONFIGURATOR = '0x8145eddDf43f50276641b55bd3AD95944510021E';
exports.ORACLE = '0x3C90887Ede8D65ccb2777A5d577beAb2548280AD';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x69FA688f1Dc47d4B5d8029D5a35FB7a548310654';
exports.ACL_MANAGER = '0xa72636CbcAa8F5FF95B2cc47F3CDEe83F3294a0B';
exports.ACL_ADMIN = '0xb2f0C5f37f4beD2cB51C44653cD5D84866BDcd2D';
exports.COLLECTOR = '0x8A020d92D6B119978582BE4d3EdFdC9F7b28BF31';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x929EC64c34a17401F460460D4B9390518E5B473e';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0xa5ba6E5EC19a1Bf23C857991c857dB62b2Aa187B';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0x81387c40EB75acB02757C1Ae55D5936E78c9dEd3';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x52A1CeB68Ee6b7B5D13E0376A1E0E4423A8cE26e';
exports.CHAIN_ID = 1666600000;
exports.EMISSION_MANAGER = '0x048f2228D7Bf6776f99aB50cB1b1eaB4D1d4cA73';
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x770ef9f4fe897e59daCc474EF11238303F9552b6';
exports.WETH_GATEWAY = '0xE387c6053CE8EC9f8C3fa5cE085Af73114a695d3';
exports.WALLET_BALANCE_PROVIDER = '0xBc790382B3686abffE4be14A030A96aC6154023a';
exports.UI_POOL_DATA_PROVIDER = '0x1DcDA4de2Bf6c7AD9a34788D22aE6b7d55016e1f';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xf7a60467aBb8A3240A0382b22E1B03c7d4F59Da5';
