// @bgd-labs/aave-address-book/dist/AaveV2Ethereum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.MIGRATION_HELPER = exports.SWAP_COLLATERAL_ADAPTER = exports.REPAY_WITH_COLLATERAL_ADAPTER = exports.WETH_GATEWAY = exports.POOL_ADDRESSES_PROVIDER_REGISTRY = exports.CHAIN_ID = exports.EMISSION_MANAGER = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.EMERGENCY_ADMIN = exports.POOL_ADMIN = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.LENDING_RATE_ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5';
exports.POOL = '0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9';
exports.POOL_CONFIGURATOR = '0x311Bb771e4F8952E6Da169b425E7e92d6Ac45756';
exports.LENDING_RATE_ORACLE = '0x8A32f49FFbA88aba6EFF96F45D8BD1D4b3f35c7D';
exports.ORACLE = '0xA50ba011c48153De246E5192C8f9258A2ba79Ca9';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d';
exports.POOL_ADMIN = '0xEE56e2B3D491590B5b31738cC34d5232F378a8D5';
exports.EMERGENCY_ADMIN = '0xCA76Ebd8617a03126B6FB84F9b1c1A0fB71C2633';
exports.COLLECTOR = '0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0xd784927Ff2f95ba542BfC824c8a8a98F3495f6b5';
exports.EMISSION_MANAGER = '0xEE56e2B3D491590B5b31738cC34d5232F378a8D5';
exports.CHAIN_ID = 1;
exports.POOL_ADDRESSES_PROVIDER_REGISTRY = '0x52D306e36E3B6B02c153d0266ff0f85d18BCD413';
exports.WETH_GATEWAY = '0xEFFC18fC3b7eb8E676dac549E0c693ad50D1Ce31';
exports.REPAY_WITH_COLLATERAL_ADAPTER = '0x80Aca0C645fEdABaa20fd2Bf0Daf57885A309FE6';
exports.SWAP_COLLATERAL_ADAPTER = '0x135896DE8421be2ec868E0b811006171D9df802A';
exports.MIGRATION_HELPER = '0xB748952c7BC638F31775245964707Bcc5DDFabFC';
exports.WALLET_BALANCE_PROVIDER = '0x8E8dAd5409E0263a51C0aB5055dA66Be28cFF922';
exports.UI_POOL_DATA_PROVIDER = '0x00e50FAB64eBB37b87df06Aa46b8B35d5f1A4e1A';
exports.UI_INCENTIVE_DATA_PROVIDER = '0xD01ab9a6577E1D84F142e44D49380e23A340387d';
