// @bgd-labs/aave-address-book/dist/AaveAddressBook.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AaveV3ScrollAlpha = exports.AaveV3OptimismGoerli = exports.AaveV3Optimism = exports.AaveV3Harmony = exports.AaveV3Fantom = exports.AaveV3FantomTestnet = exports.AaveV3ArbitrumGoerli = exports.AaveV3Arbitrum = exports.AaveV3Avalanche = exports.AaveV2Avalanche = exports.AaveV3Fuji = exports.AaveV2Fuji = exports.AaveV3Polygon = exports.AaveV2Polygon = exports.AaveV3Mumbai = exports.AaveV2Mumbai = exports.AaveV3GoerliGho = exports.AaveV3Goerli = exports.AaveV3Sepolia = exports.AaveV2Goerli = exports.AaveV2EthereumArc = exports.AaveV2EthereumAMM = exports.AaveV3Ethereum = exports.AaveV2Ethereum = exports.AaveSafetyModule = exports.AaveGovernanceV2 = void 0;
exports.AaveGovernanceV2 = __importStar(require("./AaveGovernanceV2"));
exports.AaveSafetyModule = __importStar(require("./AaveSafetyModule"));
// autogenerated entrypoints
exports.AaveV2Ethereum = __importStar(require("./AaveV2Ethereum"));
exports.AaveV3Ethereum = __importStar(require("./AaveV3Ethereum"));
exports.AaveV2EthereumAMM = __importStar(require("./AaveV2EthereumAMM"));
exports.AaveV2EthereumArc = __importStar(require("./AaveV2EthereumArc"));
exports.AaveV2Goerli = __importStar(require("./AaveV2Goerli"));
exports.AaveV3Sepolia = __importStar(require("./AaveV3Sepolia"));
exports.AaveV3Goerli = __importStar(require("./AaveV3Goerli"));
exports.AaveV3GoerliGho = __importStar(require("./AaveV3GoerliGho"));
exports.AaveV2Mumbai = __importStar(require("./AaveV2Mumbai"));
exports.AaveV3Mumbai = __importStar(require("./AaveV3Mumbai"));
exports.AaveV2Polygon = __importStar(require("./AaveV2Polygon"));
exports.AaveV3Polygon = __importStar(require("./AaveV3Polygon"));
exports.AaveV2Fuji = __importStar(require("./AaveV2Fuji"));
exports.AaveV3Fuji = __importStar(require("./AaveV3Fuji"));
exports.AaveV2Avalanche = __importStar(require("./AaveV2Avalanche"));
exports.AaveV3Avalanche = __importStar(require("./AaveV3Avalanche"));
exports.AaveV3Arbitrum = __importStar(require("./AaveV3Arbitrum"));
exports.AaveV3ArbitrumGoerli = __importStar(require("./AaveV3ArbitrumGoerli"));
exports.AaveV3FantomTestnet = __importStar(require("./AaveV3FantomTestnet"));
exports.AaveV3Fantom = __importStar(require("./AaveV3Fantom"));
exports.AaveV3Harmony = __importStar(require("./AaveV3Harmony"));
exports.AaveV3Optimism = __importStar(require("./AaveV3Optimism"));
exports.AaveV3OptimismGoerli = __importStar(require("./AaveV3OptimismGoerli"));
exports.AaveV3ScrollAlpha = __importStar(require("./AaveV3ScrollAlpha"));
