// @bgd-labs/aave-address-book/dist/AaveV3Sepolia.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UI_INCENTIVE_DATA_PROVIDER = exports.UI_POOL_DATA_PROVIDER = exports.WALLET_BALANCE_PROVIDER = exports.FAUCET = exports.WETH_GATEWAY = exports.EMISSION_MANAGER = exports.CHAIN_ID = exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = exports.DEFAULT_A_TOKEN_IMPL_REV_1 = exports.DEFAULT_INCENTIVES_CONTROLLER = exports.COLLECTOR = exports.ACL_ADMIN = exports.ACL_MANAGER = exports.AAVE_PROTOCOL_DATA_PROVIDER = exports.ORACLE = exports.POOL_CONFIGURATOR = exports.POOL = exports.POOL_ADDRESSES_PROVIDER = void 0;
// AUTOGENERATED - DON'T MANUALLY CHANGE
exports.POOL_ADDRESSES_PROVIDER = '0x0496275d34753A48320CA58103d5220d394FF77F';
exports.POOL = '0xE7EC1B0015eb2ADEedb1B7f9F1Ce82F9DAD6dF08';
exports.POOL_CONFIGURATOR = '0x9EB3fd3b5b405c29825dD5eEB7b31A08f9510397';
exports.ORACLE = '0x132C06E86CcCf93Afef7B33f0FF3e2E97EECf8f6';
exports.AAVE_PROTOCOL_DATA_PROVIDER = '0xC7832d5ACDDbdA5a7c9d1Fa9F677f9a84b7a4125';
exports.ACL_MANAGER = '0x3F7E88aD1938634150A3480ac5a58706F7a7655A';
exports.ACL_ADMIN = '0xfA0e305E0f46AB04f00ae6b5f4560d61a2183E00';
exports.COLLECTOR = '0x3246D6476dCc255b64342911D267F6CA3013068d';
exports.DEFAULT_INCENTIVES_CONTROLLER = '0x74269d0f5f1C2C44663f0214f2EE42c00a144460';
exports.DEFAULT_A_TOKEN_IMPL_REV_1 = '0x177F4611B27Cb66f5E6A51B4DD956f37A75F883B';
exports.DEFAULT_VARIABLE_DEBT_TOKEN_IMPL_REV_1 = '0xfbC748B4f6FFB9997B8dcC10Fb7494620747a82c';
exports.DEFAULT_STABLE_DEBT_TOKEN_IMPL_REV_1 = '0x61e11abA0Ea933700A74155C845b8ED6C200aCc8';
exports.CHAIN_ID = 11155111;
exports.EMISSION_MANAGER = '0x139D4efdDAaC44dfbF1Fbe9b23A635663ebfD4Af';
exports.WETH_GATEWAY = '0xe70cDC67C91d5519DD4682cA162E40480773255a';
exports.FAUCET = '0x1f885520b7BD528E46b390040F12E753Dce43004';
exports.WALLET_BALANCE_PROVIDER = '0x4f04218ECC7D2756f84836D2DC267293504cAbCc';
exports.UI_POOL_DATA_PROVIDER = '0xe38c93180a6CE4173d88793c0671AD1Dd417c7d8';
exports.UI_INCENTIVE_DATA_PROVIDER = '0x7cE1062e0688BC3D7699987A38bba84353F69881';
