// viem/dist/cjs/constants/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.weiUnits = exports.gweiUnits = exports.etherUnits = exports.solidityPanic = exports.solidityError = exports.panicReasons = exports.aggregate3Signature = exports.multicall3Abi = void 0;
var abis_js_1 = require("./abis.js");
Object.defineProperty(exports, "multicall3Abi", { enumerable: true, get: function () { return abis_js_1.multicall3Abi; } });
var contract_js_1 = require("./contract.js");
Object.defineProperty(exports, "aggregate3Signature", { enumerable: true, get: function () { return contract_js_1.aggregate3Signature; } });
var solidity_js_1 = require("./solidity.js");
Object.defineProperty(exports, "panicReasons", { enumerable: true, get: function () { return solidity_js_1.panicReasons; } });
Object.defineProperty(exports, "solidityError", { enumerable: true, get: function () { return solidity_js_1.solidityError; } });
Object.defineProperty(exports, "solidityPanic", { enumerable: true, get: function () { return solidity_js_1.solidityPanic; } });
var unit_js_1 = require("./unit.js");
Object.defineProperty(exports, "etherUnits", { enumerable: true, get: function () { return unit_js_1.etherUnits; } });
Object.defineProperty(exports, "gweiUnits", { enumerable: true, get: function () { return unit_js_1.gweiUnits; } });
Object.defineProperty(exports, "weiUnits", { enumerable: true, get: function () { return unit_js_1.weiUnits; } });
