// viem/dist/cjs/actions/public/simulateContract.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.simulateContract = void 0;
const index_js_1 = require("../../utils/index.js");
const call_js_1 = require("./call.js");
async function simulateContract(client, { abi, address, args, functionName, ...callRequest }) {
    const account = callRequest.account
        ? (0, index_js_1.parseAccount)(callRequest.account)
        : undefined;
    const calldata = (0, index_js_1.encodeFunctionData)({
        abi,
        args,
        functionName,
    });
    try {
        const { data } = await (0, call_js_1.call)(client, {
            batch: false,
            data: calldata,
            to: address,
            ...callRequest,
        });
        const result = (0, index_js_1.decodeFunctionResult)({
            abi,
            args,
            functionName,
            data: data || '0x',
        });
        return {
            result,
            request: {
                abi,
                address,
                args,
                functionName,
                ...callRequest,
            },
        };
    }
    catch (err) {
        throw (0, index_js_1.getContractError)(err, {
            abi: abi,
            address,
            args,
            docsPath: '/docs/contract/simulateContract',
            functionName,
            sender: account?.address,
        });
    }
}
exports.simulateContract = simulateContract;
