// viem/dist/cjs/utils/transaction/parseTransaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseTransaction = void 0;
const index_js_1 = require("../../errors/index.js");
const index_js_2 = require("../../errors/index.js");
const index_js_3 = require("../address/index.js");
const index_js_4 = require("../data/index.js");
const index_js_5 = require("../encoding/index.js");
const index_js_6 = require("../hash/index.js");
const assertTransaction_js_1 = require("./assertTransaction.js");
const getSerializedTransactionType_js_1 = require("./getSerializedTransactionType.js");
function parseTransaction(serializedTransaction) {
    const type = (0, getSerializedTransactionType_js_1.getSerializedTransactionType)(serializedTransaction);
    if (type === 'eip1559')
        return parseTransactionEIP1559(serializedTransaction);
    if (type === 'eip2930')
        return parseTransactionEIP2930(serializedTransaction);
    return parseTransactionLegacy(serializedTransaction);
}
exports.parseTransaction = parseTransaction;
function parseTransactionEIP1559(serializedTransaction) {
    const transactionArray = (0, index_js_5.fromRlp)(`0x${serializedTransaction.slice(4)}`, 'hex');
    const [chainId, nonce, maxPriorityFeePerGas, maxFeePerGas, gas, to, value, data, accessList, v, r, s,] = transactionArray;
    if (!(transactionArray.length === 9 || transactionArray.length === 12))
        throw new index_js_2.InvalidSerializedTransactionError({
            attributes: {
                chainId,
                nonce,
                maxPriorityFeePerGas,
                maxFeePerGas,
                gas,
                to,
                value,
                data,
                accessList,
                ...(transactionArray.length > 9
                    ? {
                        v,
                        r,
                        s,
                    }
                    : {}),
            },
            serializedTransaction,
            type: 'eip1559',
        });
    const transaction = {
        chainId: (0, index_js_5.hexToNumber)(chainId),
        type: 'eip1559',
    };
    if ((0, index_js_4.isHex)(to) && to !== '0x')
        transaction.to = to;
    if ((0, index_js_4.isHex)(gas) && gas !== '0x')
        transaction.gas = (0, index_js_5.hexToBigInt)(gas);
    if ((0, index_js_4.isHex)(data) && data !== '0x')
        transaction.data = data;
    if ((0, index_js_4.isHex)(nonce) && nonce !== '0x')
        transaction.nonce = (0, index_js_5.hexToNumber)(nonce);
    if ((0, index_js_4.isHex)(value) && value !== '0x')
        transaction.value = (0, index_js_5.hexToBigInt)(value);
    if ((0, index_js_4.isHex)(maxFeePerGas) && maxFeePerGas !== '0x')
        transaction.maxFeePerGas = (0, index_js_5.hexToBigInt)(maxFeePerGas);
    if ((0, index_js_4.isHex)(maxPriorityFeePerGas) && maxPriorityFeePerGas !== '0x')
        transaction.maxPriorityFeePerGas = (0, index_js_5.hexToBigInt)(maxPriorityFeePerGas);
    if (accessList.length !== 0 && accessList !== '0x')
        transaction.accessList = parseAccessList(accessList);
    (0, assertTransaction_js_1.assertTransactionEIP1559)(transaction);
    const signature = transactionArray.length === 12
        ? parseEIP155Signature(transactionArray)
        : undefined;
    return { ...signature, ...transaction };
}
function parseTransactionEIP2930(serializedTransaction) {
    const transactionArray = (0, index_js_5.fromRlp)(`0x${serializedTransaction.slice(4)}`, 'hex');
    const [chainId, nonce, gasPrice, gas, to, value, data, accessList, v, r, s] = transactionArray;
    if (!(transactionArray.length === 8 || transactionArray.length === 11))
        throw new index_js_2.InvalidSerializedTransactionError({
            attributes: {
                chainId,
                nonce,
                gasPrice,
                gas,
                to,
                value,
                data,
                accessList,
                ...(transactionArray.length > 8
                    ? {
                        v,
                        r,
                        s,
                    }
                    : {}),
            },
            serializedTransaction,
            type: 'eip2930',
        });
    const transaction = {
        chainId: (0, index_js_5.hexToNumber)(chainId),
        type: 'eip2930',
    };
    if ((0, index_js_4.isHex)(to) && to !== '0x')
        transaction.to = to;
    if ((0, index_js_4.isHex)(gas) && gas !== '0x')
        transaction.gas = (0, index_js_5.hexToBigInt)(gas);
    if ((0, index_js_4.isHex)(data) && data !== '0x')
        transaction.data = data;
    if ((0, index_js_4.isHex)(nonce) && nonce !== '0x')
        transaction.nonce = (0, index_js_5.hexToNumber)(nonce);
    if ((0, index_js_4.isHex)(value) && value !== '0x')
        transaction.value = (0, index_js_5.hexToBigInt)(value);
    if ((0, index_js_4.isHex)(gasPrice) && gasPrice !== '0x')
        transaction.gasPrice = (0, index_js_5.hexToBigInt)(gasPrice);
    if (accessList.length !== 0 && accessList !== '0x')
        transaction.accessList = parseAccessList(accessList);
    (0, assertTransaction_js_1.assertTransactionEIP2930)(transaction);
    const signature = transactionArray.length === 11
        ? parseEIP155Signature(transactionArray)
        : undefined;
    return { ...signature, ...transaction };
}
function parseTransactionLegacy(serializedTransaction) {
    const transactionArray = (0, index_js_5.fromRlp)(serializedTransaction, 'hex');
    const [nonce, gasPrice, gas, to, value, data, chainIdOrV_, r, s] = transactionArray;
    if (!(transactionArray.length === 6 || transactionArray.length === 9))
        throw new index_js_2.InvalidSerializedTransactionError({
            attributes: {
                nonce,
                gasPrice,
                gas,
                to,
                value,
                data,
                ...(transactionArray.length > 6
                    ? {
                        v: chainIdOrV_,
                        r,
                        s,
                    }
                    : {}),
            },
            serializedTransaction,
            type: 'legacy',
        });
    const transaction = {
        type: 'legacy',
    };
    if ((0, index_js_4.isHex)(to) && to !== '0x')
        transaction.to = to;
    if ((0, index_js_4.isHex)(gas) && gas !== '0x')
        transaction.gas = (0, index_js_5.hexToBigInt)(gas);
    if ((0, index_js_4.isHex)(data) && data !== '0x')
        transaction.data = data;
    if ((0, index_js_4.isHex)(nonce) && nonce !== '0x')
        transaction.nonce = (0, index_js_5.hexToNumber)(nonce);
    if ((0, index_js_4.isHex)(value) && value !== '0x')
        transaction.value = (0, index_js_5.hexToBigInt)(value);
    if ((0, index_js_4.isHex)(gasPrice) && gasPrice !== '0x')
        transaction.gasPrice = (0, index_js_5.hexToBigInt)(gasPrice);
    (0, assertTransaction_js_1.assertTransactionLegacy)(transaction);
    if (transactionArray.length === 6)
        return transaction;
    const chainIdOrV = (0, index_js_4.isHex)(chainIdOrV_) && chainIdOrV_ !== '0x'
        ? (0, index_js_5.hexToBigInt)(chainIdOrV_)
        : 0n;
    if (s === '0x' && r === '0x') {
        if (chainIdOrV > 0)
            transaction.chainId = Number(chainIdOrV);
        return transaction;
    }
    const v = chainIdOrV;
    const chainId = Number((v - 35n) / 2n);
    if (chainId > 0)
        transaction.chainId = chainId;
    else if (v !== 27n && v !== 28n)
        throw new index_js_2.InvalidLegacyVError({ v });
    transaction.v = v;
    transaction.s = s;
    transaction.r = r;
    return transaction;
}
function parseAccessList(accessList_) {
    const accessList = [];
    for (let i = 0; i < accessList_.length; i++) {
        const [address, storageKeys] = accessList_[i];
        if (!(0, index_js_3.isAddress)(address))
            throw new index_js_1.InvalidAddressError({ address });
        accessList.push({
            address: address,
            storageKeys: storageKeys.map((key) => ((0, index_js_6.isHash)(key) ? key : (0, index_js_4.trim)(key))),
        });
    }
    return accessList;
}
function parseEIP155Signature(transactionArray) {
    const signature = transactionArray.slice(-3);
    const v = signature[0] === '0x' || (0, index_js_5.hexToBigInt)(signature[0]) === 0n ? 27n : 28n;
    return {
        r: (0, index_js_4.padHex)(signature[1], { size: 32 }),
        s: (0, index_js_4.padHex)(signature[2], { size: 32 }),
        v,
        yParity: v === 27n ? 0 : 1,
    };
}
